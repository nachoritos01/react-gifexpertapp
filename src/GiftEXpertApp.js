import React , {useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GiftGrid } from './components/GiftGrid';

export const GiftEXpertApp = ( {defaultCategories =['One puch']} ) => {
    // const categories = ['One puch', 'Samurai x', 'Dragon ball']
    const [categories, setCategories] = useState(defaultCategories);

    /* const handleAdd = ( ) =>{
        setCategories(['HunterxHunter', ...categories]);
    } */

    return (
        <div>
            <h2> GifExperrtApp </h2>
            <hr></hr>
            <AddCategory setCategories={setCategories} />
            <ol>
                {
                    categories.map( category => <GiftGrid key={category} category={category} />)
                }
            </ol>
        </div>
    )
}