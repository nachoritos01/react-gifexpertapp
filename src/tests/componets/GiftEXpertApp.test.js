import React from 'react'
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GiftEXpertApp } from '../../GiftEXpertApp';


describe('Probando el componente <GiftEXpertApp/>', () => {

    
    test('Deberia de mostrarse el componente correctamente ', () => {
        const wrapper = shallow(<GiftEXpertApp/>);
        
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Deberia de mostrar una lista de categorias ', () => {
        const categories = [ 'One puch men', 'Dragon ball' ];
        const wrapper = shallow(<GiftEXpertApp defaultCategories={categories}/>);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GiftGrid').length).toBe(categories.length);
    });
    
    
})

