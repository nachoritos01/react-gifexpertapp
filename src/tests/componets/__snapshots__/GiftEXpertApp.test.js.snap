// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <GiftEXpertApp/> Deberia de mostrar una lista de categorias  1`] = `
<div>
  <h2>
     GifExperrtApp 
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GiftGrid
      category="One puch men"
      key="One puch men"
    />
    <GiftGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</div>
`;

exports[`Probando el componente <GiftEXpertApp/> Deberia de mostrarse el componente  1`] = `
<div>
  <h2>
     GifExperrtApp 
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GiftGrid
      category="One puch"
      key="One puch"
    />
  </ol>
</div>
`;

exports[`Probando el componente <GiftEXpertApp/> Deberia de mostrarse el componente correctamente  1`] = `
<div>
  <h2>
     GifExperrtApp 
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GiftGrid
      category="One puch"
      key="One puch"
    />
  </ol>
</div>
`;
